zcash_client_backend/src/data_api.rs

```diff
@@ -159,21 +167,34 @@ pub trait WalletRead {
     /// with which they are associated.
     fn get_nullifiers(&self) -> Result<Vec<(AccountId, Nullifier)>, Self::Error>;
 
-    /// Return all spendable notes.
-    fn get_spendable_notes(
+    /// Returns all nullifiers (including those for notes that have been previously spent), along
+    /// with the account identifiers with which they are associated.
+    fn get_all_nullifiers(&self) -> Result<Vec<(AccountId, Nullifier)>, Self::Error>;
+
+    /// Return all unspent Sapling notes.
+    fn get_spendable_sapling_notes(
         &self,
         account: AccountId,
         anchor_height: BlockHeight,
     ) -> Result<Vec<SpendableNote>, Self::Error>;
 
-    /// Returns a list of spendable notes sufficient to cover the specified
+    /// Returns a list of spendable Sapling notes sufficient to cover the specified
     /// target value, if possible.
-    fn select_spendable_notes(
+    fn select_spendable_sapling_notes(
         &self,
         account: AccountId,
         target_value: Amount,
         anchor_height: BlockHeight,
     ) -> Result<Vec<SpendableNote>, Self::Error>;
+
+    #[cfg(feature = "transparent-inputs")]
+    /// Returns a list of unspent transparent UTXOs that appear in the chain at heights up to and
+    /// including `max_height`.
+    fn get_unspent_transparent_outputs(
```

Comment:
has the visibility of this function changed?
![image](https://user-images.githubusercontent.com/484008/150538376-68a328c7-d3f4-411b-b7e1-bf44b100f9fa.png)